HLS in partially broken

The HLS works for `kore-rpc-types`, `kore` and `dev-tools`, but only seem to work partially in `hs-backend-booster`.  Any of the Booster files that transitively import the LLVM bindings do not have HLS goodies.

I think I've isolated the issue to the following command that could be run from the Nix shell:
```
 haskell-language-server-wrapper --debug booster/library/Booster/LLVM/Internal.hs 2>&1 | tee hls.log
```

I eventually  get
```
...
  Exception when trying to run compile-time code:
  ./cbits/kllvm-c.h: openBinaryFile: does not exist (No such file or directory)
  Code: (dynamicBindings "./cbits/kllvm-c.h")
```

I seem to have reduced the reproduction steps to the `cabal repl` command, which is called internally by HLS: 
```
cabal --builddir=/home/geo2a/.cache/hie-bios/dist-pr-fbb12002ac4eebc87aa7ba088993096e v2-repl \
          --with-compiler /home/geo2a/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36   \
          --with-hc-pkg /home/geo2a/.cache/hie-bios/ghc-pkg-6f91fcd580661d243f35998a81beb66d lib:hs-backend-booster
```
which produces
```
/nix/store/0gi4vbw1qfjncdl95a9ply43ymd6aprm-binutils-2.40/bin/ld.gold: fatal error: .: pread failed: Is a directory
collect2: error: ld returned 1 exit status
`cc' failed in phase `Linker'. (Exit code: 1)
Error: cabal: Failed to build kore-rpc-types-0.60.0.0 (which is required by
hs-backend-booster-0.0.0).
```

I've tried varying the three arguments, and the `--with-compiled`  seems to be the problem. If I remove this argument or substitute it with `--with-compiler ghc`, the command successfully starts the repl.

The exact command will be in `hls.log` produced by
```
 haskell-language-server-wrapper --debug booster/library/Booster/LLVM/Internal.hs 2>&1 | tee hls.log
```

Could it be that we somehow have to slightly different copies of GHC `9.6.4` in the flake, and HLS somehow uses the wrong one?

--- a/.gitignore
+++ b/.gitignore
@@ -59,3 +59,6 @@ booster/*.cabal
 /booster/scripts/bug-reports
 /booster/test/rpc-integration/resources/*.dylib
 /booster/test/*/definition/*kompiled/
+
+# the LLVM bindings from ./booster/cbits are symlinked to the root ./cbits to make HLS work
+/cbits

